cpp/modmesh/universe/polygon.hpp

```diff
@@ -781,6 +788,617 @@ class TrianglePad
 using TrianglePadFp32 = TrianglePad<float>;
 using TrianglePadFp64 = TrianglePad<double>;
 
+/**
+ * Forward declaration of PolygonPad for use in Polygon3d handle class.
+ */
+template <typename T>
+class PolygonPad;
+
+/**
+ * Polygon3d handle class - lightweight view into a polygon stored in PolygonPad.
+ *
+ * This is a non-owning handle that references a polygon stored in a PolygonPad
+ * container. Polygons are defined by an ordered list of nodes following the
+ * right-hand rule: counter-clockwise for positive area, clockwise for negative.
+ *
+ * The handle uses polygon_id as public API, with internal offset/count for efficient access.
+ *
+ * @tparam T floating-point type
+ */
+template <typename T>
+class Polygon3d
+{
+
+public:
+
+    using point_type = Point3d<T>;
+    using value_type = T;
+    using segment_type = Segment3d<T>;
+    using polygon_pad_type = PolygonPad<T>;
+
+    Polygon3d(std::shared_ptr<polygon_pad_type> pad, size_t polygon_id)
+        : m_pad(std::move(pad))
+        , m_id(polygon_id)
+    {
+    }
+
+    Polygon3d() = default;
+    Polygon3d(Polygon3d const &) = default;
+    Polygon3d(Polygon3d &&) = default;
+    Polygon3d & operator=(Polygon3d const &) = default;
+    Polygon3d & operator=(Polygon3d &&) = default;
+    ~Polygon3d() = default;
+
+    size_t polygon_id() const { return m_id; }
+    std::shared_ptr<polygon_pad_type> pad() const { return m_pad; }
+
+    size_t nnode() const
+    {
+        if (m_id >= m_pad->m_begins.size())
+        {
+            throw std::out_of_range(
+                std::format("Polygon3d::nnode: polygon_id {} >= num_polygons {}",
+                            m_id,
+                            m_pad->m_begins.size()));
+        }
+        typename polygon_pad_type::ssize_type const begin_index = m_pad->m_begins[m_id];
+        typename polygon_pad_type::ssize_type const end_index = m_pad->m_ends[m_id];
+        return static_cast<size_t>(end_index - begin_index);
+    }
+
+    uint8_t ndim() const
+    {
+        return m_pad->ndim();
+    }
+
+    point_type node(size_t index) const
```

Comment:
Most code in `Polygon3d::node()` should be moved to `PolygonPad()` and the former just calls the latter.